projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.spec.ts

```diff
@@ -0,0 +1,312 @@
+
+      this.displayInputVal = inputVal;
+      this.propType = this.displayInputVal.property;
+      this.parentResource = res[0];
+      this.mode = 'read';
+    });
+
+  }
+}
+
+/**
+ * Test host component to simulate parent component.
+ */
+@Component({
+  template: `
+    <kui-link-value #inputVal [mode]="mode" [parentResource]="parentResource" [propType]="propType"></kui-link-value>`
+})
+class TestHostCreateValueComponent implements OnInit {
+
+  @ViewChild('inputVal', {static: false}) inputValueComponent: LinkValueComponent;
+  parentResource: ReadResource;
+  propType: string;
+  mode: 'read' | 'update' | 'create' | 'search';
+
+  ngOnInit() {
+
+    MockResource.getTestthing().subscribe(res => {
+      this.propType = 'http://0.0.0.0:3333/ontology/0001/anything/v2#hasOtherThingValue';
+      this.parentResource = res[0];
+      this.mode = 'create';
+    });
+  }
+}
+
+describe('LinkValueComponent', () => {
+
+  beforeEach(async(() => {
+    const valuesSpyObj = {
+      v2: {
+        values: jasmine.createSpyObj('values', ['updateValue', 'getValue']),
+        search: jasmine.createSpyObj('search', ['doSearchByLabel']),
+      }
+    };
+    TestBed.configureTestingModule({
+      declarations: [
+        LinkValueComponent,
+        TestHostDisplayValueComponent,
+        TestHostCreateValueComponent
+       ],
+       imports: [
+        ReactiveFormsModule,
+        MatInputModule,
+        MatAutocompleteModule,
+        BrowserAnimationsModule
+      ],
+      providers: [
+        {
+          provide: KnoraApiConnectionToken,
+          useValue: valuesSpyObj
+        }
+      ]
+    })
+    .compileComponents();
+  }));
+
+  describe('display and edit a link value', () => {
+    let testHostComponent: TestHostDisplayValueComponent;
+    let testHostFixture: ComponentFixture<TestHostDisplayValueComponent>;
+    let valueComponentDe: DebugElement;
+    let valueInputNativeElement;
+    let commentInputDebugElement: DebugElement;
+    let commentInputNativeElement;
+
+    beforeEach(() => {
+      testHostFixture = TestBed.createComponent(TestHostDisplayValueComponent);
+      testHostComponent = testHostFixture.componentInstance;
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent).toBeTruthy();
+      expect(testHostComponent.inputValueComponent).toBeTruthy();
+
+      const hostCompDe = testHostFixture.debugElement;
+
+      valueComponentDe = hostCompDe.query(By.directive(LinkValueComponent));
+      valueInputNativeElement = valueComponentDe.query(By.css('input')).nativeElement;
+
+      commentInputDebugElement = valueComponentDe.query(By.css('input.comment'));
+      commentInputNativeElement = commentInputDebugElement.nativeElement;
+
+    });
+
+    it('should display an existing value', () => {
+
+      expect(testHostComponent.displayInputVal.linkedResourceIri).toMatch('http://rdfh.ch/0001/0C-0L1kORryKzJAJxxRyRQ');
```

Comment:
Was your intention to check for `inputValueComponent.displayValue`?

Reply:
the intention was to check the iri of the linkedresource.

Reply:
you mean on the test component?

Reply:
I think you are checking the value on the mocked parent, not on the component that ought to be tested here.